Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyFilePathCtrl.cpp

```diff
@@ -26,6 +26,61 @@ AZ_POP_DISABLE_WARNING
 
 namespace AzToolsFramework
 {
+    AZ::IO::Path GetAbsolutePathFromRelativePath(const AZ::IO::Path& relativePath, const AZStd::string& productExtension)
```

Comment:
If this control is intended for selecting source file paths, does it need a product path and extension? You might be able to replace this whole function with

```
            /**
            * Given a path to a source file, retrieve its actual watch folder path and details.
            * @param sourcePath is either a relative or absolute path to a source file.
            * @param assetInfo is a /ref AZ::Data::AssetInfo filled out with details about the asset including its relative path to its watch folder
            *   note that inside assetInfo is a AssetId, but only the UUID-part will ever have a value since we are dealing with a source file (no subid)
            * @param watchFolder is the scan folder that it was found inside (the path in the assetInfo is relative to this folder).
            *   If you Path::Join the watchFolder and the assetInfo relative path, you get the full path.
            * returns false if it cannot find the source, true otherwise.
            */
            virtual bool GetSourceInfoBySourcePath(const char* sourcePath, AZ::Data::AssetInfo& assetInfo, AZStd::string& watchFolder) = 0;
```

Reply:
That API only works if the path exists on disk, which doesn't work in this case because we want to be able to specify a new file that doesn't exist yet.

Reply:
Discussed offline, we should be able to use this API for the "path exists" case which will eliminate the need to provide the product extension.